### PR TITLE
fix(ci): add back unit test and add cargo check release

### PR DESCRIPTION
**Pull request by huangjw806**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- back unit test to the main pipeline to solve codecov behaves strangely 
- add cargo check release mod in pr pipeline 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#6277

### CODE REVIEW

File: ci/workflows/main.yml [hunk 1]
```diff
@@ -186,6 +186,22 @@ steps:
...
+      - gencer/cache#v2.4.10: *cargo-cache
+      - seek-oss/aws-sm#v2.3.1:
+          env:
+            CODECOV_TOKEN: my-codecov-token
+      - docker-compose#v3.9.0:
+          run: rw-build-env
+          config: ci/docker-compose.yml
+          environment:
+            - CODECOV_TOKEN
+    timeout_in_minutes: 15
```
Comment (StrikeW) [inline ci/workflows/main.yml:202]: Have you test that 15mins is enough to finish?

File: ci/scripts/pre-check.sh [hunk 2]
```diff
@@ -5,11 +5,13 @@ set -euo pipefail
 
 source ci/scripts/common.env.sh
 
+echo "--- Check release mod"
+cargo check --release --features "static-link static-log-level"
```
Comment (BugenZhao) [inline ci/scripts/pre-check.sh:9]: Let's still run `cargo clippy` here like the following 2 lines. `cargo clippy` contains all the content of `cargo check`.

File: ci/scripts/pre-check.sh [hunk 3]
```diff
@@ -5,11 +5,13 @@ set -euo pipefail
...
+
 echo "--- Run clippy check"
 cargo clippy --all-targets --all-features --locked -- -D warnings
 
 echo "--- Build documentation"
 cargo doc --document-private-items --no-deps
 
 echo "--- Run doctest"
-cargo test --doc
```
Comment (BugenZhao) [inline ci/scripts/pre-check.sh:15]: Recover this line? 👀

File: ci/workflows/pull-request.yml [hunk 4]
```diff
@@ -170,8 +170,8 @@ steps:
     timeout_in_minutes: 15
     retry: *auto-retry
 
-  - label: "pre unit test"
-    command: "ci/scripts/pre-unit-test.sh"
+  - label: "pre check"
+    command: "ci/scripts/pre-check.sh"
```
Comment (BugenZhao) [inline ci/workflows/pull-request.yml:174]: How about simply naming it `check`?

File: ci/workflows/main.yml (hunk 1 above)
Comment (huangjw806) [reply ci/workflows/main.yml:202]: yes, usually only seven or eight minutes.

**Review by BugenZhao — APPROVED** [review]

**Review by wenym1 — APPROVED** [review]

**Pull request merged**